src/libraries/Common/src/System/Security/Cryptography/X509Certificates/X509CertificateLoader.Pkcs12.cs

```diff
@@ -0,0 +1,944 @@
+                AsnValueReader reader = outer.ReadSequence();
+                outer.ThrowIfNotEmpty();
+
+                ReadOnlyMemory<byte> rebind = pfxAsn.AuthSafe.Content;
+                bagState.Init(loaderLimits);
+
+                int? workRemaining = loaderLimits.TotalKdfIterationLimit;
+
+                while (reader.HasData)
+                {
+                    ContentInfoAsn.Decode(ref reader, rebind, out ContentInfoAsn safeContentsAsn);
+
+                    ReadOnlyMemory<byte> contentData;
+
+                    if (safeContentsAsn.ContentType == Oids.Pkcs7Data)
+                    {
+                        contentData = Helpers.DecodeOctetStringAsMemory(safeContentsAsn.Content);
+                    }
+                    else if (safeContentsAsn.ContentType == Oids.Pkcs7Encrypted)
+                    {
+                        if (loaderLimits.IgnoreEncryptedAuthSafes)
+                        {
+                            continue;
+                        }
+
+                        bagState.PrepareDecryptBuffer(authSafeContents.Length);
+
+                        if (!bagState.LockedPassword)
+                        {
+                            bagState.LockPassword();
+                            int? workRemainingSave = workRemaining;
+
+                            try
+                            {
+                                contentData = DecryptSafeContents(
+                                    safeContentsAsn,
+                                    loaderLimits,
+                                    password,
+                                    ref bagState,
+                                    ref workRemaining);
+                            }
+                            catch (CryptographicException)
+                            {
+                                password = password.ContainsNull() ? "".AsSpan() : default;
+                                workRemaining = workRemainingSave;
+
+                                contentData = DecryptSafeContents(
+                                    safeContentsAsn,
+                                    loaderLimits,
+                                    password,
+                                    ref bagState,
+                                    ref workRemaining);
+                            }
+                        }
+                        else
+                        {
+                            contentData = DecryptSafeContents(
+                                safeContentsAsn,
+                                loaderLimits,
+                                password,
+                                ref bagState,
+                                ref workRemaining);
+                        }
+                    }
+                    else
+                    {
+                        // Should there be an option here to preserve this?
+                        // Ignore this?
```

Comment:
Yeah, I think is a place where I was previously on the side of "we shouldn't block it", but then realized it's "there's a whole chunk here that we just don't know how to read".  It might be worth making a dedicated message for, but I don't think we should skip it (without an option), otherwise there's the oddity of in some theoretical future version a file throws for being over work limits that previously worked (because it skipped this block).

So, I'm starting with removing the comment, but am open to discussion.

Reply:
I think removing the comment is fine enough. It's the behavior we have now.